File: app/src/main/java/ru/androidacademy/ateam/model/dao/DeckDao.java
package ru.androidacademy.ateam.model.dao;

import android.arch.persistence.room.*;
import android.support.annotation.NonNull;
import io.reactivex.Single;
import ru.androidacademy.ateam.model.tables.Deck;

import java.util.ArrayList;
import java.util.List;

@Dao
public interface DeckDao {

    @Query("SELECT * FROM deck")
    Single<List<Deck>> getAllDeck();

    @Query("SELECT COUNT(*) FROM deck")
    int getAllDeckCount();

    @Query("SELECT * FROM deck WHERE deckId = :deckId")
    Single<List<Deck>> getDeckById(@NonNull long deckId);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Deck deck);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertDeckList(List<Deck> decks);

    @Update
    void update(Deck deck);

    @Delete
    void delete(Deck deck);
}

File: app/src/main/java/ru/androidacademy/ateam/db/dao/DeckDao.kt
package ru.androidacademy.ateam.db.dao

import androidx.room.*
import io.reactivex.Completable
import io.reactivex.Single
import ru.androidacademy.ateam.model.Deck


@Dao
interface DeckDao {

    @get:Query("SELECT * FROM deck")
    val getAll: Single<List<Deck>>

    @get:Query("SELECT COUNT(*) FROM deck")
    val allDeckCount: Int

    @Query("SELECT * FROM deck WHERE id = :deckId")
    fun getDeckById(deckId: Long): Single<List<Deck>>

    @Query("SELECT * FROM deck WHERE name = :deckName")
    fun getDeckByName( deckName: String): Single<List<Deck>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(deck: Deck):Completable

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertDeckList(decks: List<Deck>):Completable

    @Update
    fun update(deck: Deck)

    @Delete
    fun delete(deck: Deck)

}
